crates/blockchain/blockchain.rs

```diff
@@ -117,16 +117,35 @@ impl Blockchain {
 
     pub fn add_block(&self, block: &Block) -> Result<(), ChainError> {
         let since = Instant::now();
+        let execution_result = self.execute_block(block);
+        let executed = Instant::now();
+        let result = execution_result.and_then(|res| self.store_block(block, res));
+        let stored = Instant::now();
 
-        let result = self
-            .execute_block(block)
-            .and_then(|res| self.store_block(block, res));
-
-        let interval = Instant::now().duration_since(since).as_millis();
+        let interval = stored.duration_since(since).as_millis();
         if interval != 0 {
             let as_gigas = (block.header.gas_used as f64).div(10_f64.powf(9_f64));
             let throughput = (as_gigas) / (interval as f64) * 1000_f64;
-            info!("[METRIC] BLOCK EXECUTION THROUGHPUT: {throughput} Gigagas/s TIME SPENT: {interval} msecs");
+            let execution_time = executed.duration_since(since).as_millis();
```

Comment:
would be nice to have this code outside the add_block function so we don't pollute it with metrics.

Reply:
Moved to separate function